plans/phase-1/multigrescluster-cr-definition.md

```diff
@@ -0,0 +1,713 @@
+        resources:
+          requests:
+            storage: "10Gi"
+    # Optional. Use this if 'managed' is false or to override the managed spec.
+    # external:
+    #   address: "my-external-etcd-client.etcd.svc:2379"
+
+  # admin provides a template for the MultiAdmin component.
+  # Optional.
+  admin:
+    # If 'managed' is true, the operator will create and manage the MultiAdmin CR.
+    # Optional. Defaults to 'true'.
+    managed: true
+    spec:
+      image: multigres/multigres:latest
+      replicas: 1
+      resources:
+        requests:
+          cpu: "100m"
+          memory: "128Mi"
+        limits:
+          cpu: "200m"
+          memory: "256Mi"
+
+  # orchestrator provides a template for the MultiOrch component.
+  # Optional.
+  orchestrator:
+    # If 'managed' is true, the operator will create and manage the MultiOrch CR.
+    # Optional. Defaults to 'true'.
+    managed: true
+    spec:
+      image: multigres/multigres:latest
+      replicas: 1
+      resources:
+        requests:
+          cpu: "100m"
+          memory: "128Mi"
+        limits:
+          cpu: "200m"
+          memory: "256Mi"
+
+  # cells contains the templates for the MultiCell CRs.
+  # Required.
+  cells:
+    # If 'managed' is true, the operator will create and manage MultiCell CRs from this list.
+    # Optional. Defaults to 'true'.
+    managed: true
+    # Defines the default image for all multiGateway instances created from these templates.
+    # This can be overridden within a specific template's spec if needed.
+    multiGatewayImage: multigres/multigres:latest
+    templates:
+      - name: "us-east-1"
+        spec:
+          multiGateway:
+            replicas: 2
+            resources:
+              requests:
+                cpu: "500m"
+                memory: "512Mi"
+              limits:
+                cpu: "1"
+                memory: "1Gi"
+          # topoServer defines a cell-local topology server.
+          # Optional. Default is to use global. In the future this can use the topology CR to create local topology
+          topoServer:
+            external:
+              address: "etcd-us-east-1.my-domain.com:2379"
+              rootPath: "/multigres/us-east-1"
+
+      - name: "us-west-2"
+        spec:
+          multiGateway:
+            replicas: 2
+            resources:
+              requests:
+                cpu: "500m"
+                memory: "512Mi"
+              limits:
+                cpu: "1"
+                memory: "1Gi"
+
+  # databases contains the templates for the MultiShard CRs.
+  # Optional.
+  databases:
+    # If 'managed' is true, the operator will create and manage MultiShard CRs from this list.
+    # Optional. Defaults to 'true'.
+    managed: true
+    templates:
+      - name: "users_db"
+        spec:
+          images:
+            postgres: postgres:15.3
+            multiPooler: multigres/multigres:latest
+          pools:
+            - cell: "us-east-1"
```

Comment:
why not put this under `cells[0].databases`?

Reply:
> not entirely sure it is a good idea to give the option to have different versions of pg and multipooler in different cells

I agree and I'm not suggesting this. To be clear, I'm also not suggesting that you entirely eliminate the `spec.databases` attribute.

With the current structure, there are 2-3 arrays where the items in those arrays contain a `cell` identifier attribute. The value of those `cell` identifier attributes need to match. This creates the opportunity for typos. By consolidating the `cell` arrays into one single array, the opportunity for mistakes is reduced.

With all cell-specific information in one place, the pg and multipooler versions can still be specified in `spec.databases` in a way which applies to all cells.